protocol/messenger.go

```diff
@@ -5723,3 +5897,79 @@ func (m *Messenger) handleSyncClearHistory(state *ReceivedMessageState, message
 	})
 	return nil
 }
+
+func (m *Messenger) handleSyncTrustedUser(state *ReceivedMessageState, message protobuf.SyncTrustedUser) error {
+	updated, err := m.verificationDatabase.UpsertTrustStatus(message.Id, verification.TrustStatus(message.Status), message.Clock)
+	if err != nil {
+		return err
+	}
+
+	if updated {
+		state.AllTrustStatus[message.Id] = verification.TrustStatus(message.Status)
+
+		contact, ok := m.allContacts.Load(message.Id)
+		if !ok {
+			m.logger.Info("contact not found")
+			return nil
+		}
+
+		contact.TrustStatus = verification.TrustStatus(message.Status)
+		m.allContacts.Store(contact.ID, contact)
+		state.ModifiedContacts.Store(contact.ID, true)
+	}
+
+	return nil
+}
+
+func ToVerificationRequest(message protobuf.SyncVerificationRequest) *verification.Request {
+	return &verification.Request{
+		From:          message.From,
+		To:            message.To,
+		Challenge:     message.Challenge,
+		Response:      message.Response,
+		RequestedAt:   message.RequestedAt,
+		RepliedAt:     message.RepliedAt,
+		RequestStatus: verification.RequestStatus(message.VerificationStatus),
+	}
+}
+
+func (m *Messenger) handleSyncVerificationRequest(state *ReceivedMessageState, message protobuf.SyncVerificationRequest) error {
+	verificationRequest := ToVerificationRequest(message)
+
+	shouldSync, err := m.verificationDatabase.UpsertVerificationRequest(verificationRequest)
+	if err != nil {
+		return err
+	}
+
+	myPubKey := hexutil.Encode(crypto.FromECDSAPub(&m.identity.PublicKey))
+
+	if !shouldSync {
+		return nil
+	}
+
+	state.AllVerificationRequests = append(state.AllVerificationRequests, verificationRequest)
+
+	if message.From == myPubKey { // Verification requests we sent
+		contact, ok := m.allContacts.Load(message.To)
+		if !ok {
+			m.logger.Info("contact not found")
+			return nil
+		}
+
+		contact.VerificationStatus = VerificationStatus(message.VerificationStatus)
+		if err := m.persistence.SaveContact(contact, nil); err != nil {
+			return err
+		}
+
+		m.allContacts.Store(contact.ID, contact)
+		state.ModifiedContacts.Store(contact.ID, true)
+
+		// TODO: create activity center notif
```

Comment:
Are these yet to be done as part of this PR or is has this been put on TODO on purpose for now?

Reply:
Activity center stuff will be done in another PR